### PR TITLE
SSL Tutorial updated as being notebooks

### PR DESCRIPTION
**Pull request by finalelement**
Signed-off-by: Vishwesh Nath <vnath@nvidia.com>

Converts the python scripts to being jupyter notebooks that are reflected in MONAI Toolkit.

### CODE REVIEW

**Comment by mingxin-zheng** [comment]
Thank you @finalelement for the PR!

- Regarding the license header, we slightly modify the template to fit Jupyter notebooks. Please copy it from [here](https://github.com/Project-MONAI/tutorials/blob/main/CONTRIBUTING.md#add-license) for the exact format.
- The Contributing guidelines require "## Setup environment" and "Setup imports" cells to standardize the notebook, and ease the packing of the toolkit. Can we ensure the notebooks following the [guidelines here](https://github.com/Project-MONAI/tutorials/blob/main/CONTRIBUTING.md#create-a-notebook)?
- Because there is no automatic download of the dataset, the notebooks needs to be excluded from the CI/CD papermill test. Can you refer to [here](https://github.com/Project-MONAI/tutorials/blob/main/CONTRIBUTING.md#notebook-execution) for  how to fix it?

**Comment by finalelement** [comment]
@mingxin-zheng I've tried addressing all the comments. Please see if there is anything more that I can do from my side.

File: self_supervised_pretraining/ssl_finetune.ipynb [hunk 1]
```diff
@@ -5,23 +5,50 @@
...
+    "!python -c \"import matplotlib\" || pip install -q matplotlib\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "88f7f858",
+   "metadata": {},
+   "source": [
+    "## Import statements"
```
Comment (mingxin-zheng) [inline self_supervised_pretraining/ssl_finetune.ipynb:51]: Can you change this to “## Setup imports" as other notebooks? Thanks!

File: self_supervised_pretraining/ssl_finetune.ipynb [hunk 2]
```diff
@@ -5,23 +5,50 @@
    "id": "14fc43c2",
    "metadata": {},
    "source": [
+    "# Finetune 3D Multi-Class Abdominal Segmentation Model Using SSL Pre-trained Weights\n",
```
Comment (mingxin-zheng) [inline self_supervised_pretraining/ssl_finetune.ipynb:8]: The license header is usually placed at the top. @finalelement may I ask you to put the title `# Finetune 3D Multi-Class Abdominal Segmentation Model Using SSL Pre-trained Weights` below the license header as the [example](https://github.com/Project-MONAI/tutorials/blob/main/.github/contributing_templates/notebook/example_feature.ipynb)? Thanks!

File: self_supervised_pretraining/ssl_finetune.ipynb (hunk 1 above)
Comment (mingxin-zheng) [reply self_supervised_pretraining/ssl_finetune.ipynb:51]: Also it misses the following required check:
```
from monai.config import print_config
....
print_config()`
```

File: self_supervised_pretraining/ssl_train.ipynb [hunk 3]
```diff
@@ -5,23 +5,51 @@
...
+   "id": "95fc8d67",
+   "metadata": {},
+   "source": [
     "##### Note:\n",
     "[TCIA Covid 19](https://wiki.cancerimagingarchive.net/display/Public/CT+Images+in+COVID-19) dataset needs to be downloaded for the execution of this notebook, both datasets need to be downloaded. Downloading data to the `data_root` path defined variable is optimal, if data is downloaded at another location please change the path `data_root` accordingly.\n",
     "\n",
     "Please see the Readme for more details about the dataset. Data splits have already been provided.\n",
     "\n",
-    "##### Import statements"
+    "## Setup Environment"
```
Comment (mingxin-zheng) [inline self_supervised_pretraining/ssl_train.ipynb:32]: A minor change is needed: "## Setup Environment" -> "Setup environment"

File: self_supervised_pretraining/ssl_train.ipynb [hunk 4]
```diff
@@ -5,23 +5,51 @@
...
+    "!python -c \"import matplotlib\" || pip install -q matplotlib\n",
+    "%matplotlib inline"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "49070e05",
+   "metadata": {},
+   "source": [
+    "## Setup Imports"
```
Comment (mingxin-zheng) [inline self_supervised_pretraining/ssl_train.ipynb:52]: The "Setup imports" also needs to `print_config()` according to the guidelines. Thanks!

**Comment by mingxin-zheng** [comment]
The comments above should fix these relevant check failures in `build/guidelines`:

- Error: Missing the "Setup imports" after the first code cell of file: /home/runner/work/tutorials/tutorials/self_supervised_pretraining/ssl_finetune.ipynb.
- Error: print_config() cannot be found after the "Setup imports" markdown cell in file: /home/runner/work/tutorials/tutorials/self_supervised_pretraining/ssl_finetune.ipynb.
- Error: Missing the "Setup imports" after the first code cell of file: /home/runner/work/tutorials/tutorials/self_supervised_pretraining/ssl_train.ipynb.
- Error: print_config() cannot be found after the "Setup imports" markdown cell in file: /home/runner/work/tutorials/tutorials/self_supervised_pretraining/ssl_train.ipynb.

But there are some errors in `build/pep8` that needs attention:
```
Checking PEP8 compliance...
stdin:290:13: B007 Loop control variable 'step' not used within the loop body. If this is intended, start the name with an underscore.
        for step, batch in enumerate(epoch_iterator_val):
            ^
Error: Try running with autofixes: --autofix.

Check failed!
Running ./self_supervised_pretraining/ssl_train.ipynb
Checking PEP8 compliance...
stdin:88:10: B007 Loop control variable 'each_d' not used within the loop body. If this is intended, start the name with an underscore.
for idx, each_d in enumerate(train_data):
         ^
stdin:91:10: B007 Loop control variable 'each_d' not used within the loop body. If this is intended, start the name with an underscore.
for idx, each_d in enumerate(val_data):
         ^
Error: Try running with autofixes: --autofix.
```

**Comment by mingxin-zheng** [comment]
Because the SSL notebooks are created in this PR, the [notebook list in `README.md`](https://github.com/Project-MONAI/tutorials/blob/main/README.md#4-list-of-notebooks-and-examples) should also be updated.

File: self_supervised_pretraining/ssl_finetune.ipynb (hunk 1 above)
Comment (finalelement) [reply self_supervised_pretraining/ssl_finetune.ipynb:51]: Addressed

File: self_supervised_pretraining/ssl_finetune.ipynb (hunk 2 above)
Comment (finalelement) [reply self_supervised_pretraining/ssl_finetune.ipynb:8]: Good catch, addressed.

File: self_supervised_pretraining/ssl_train.ipynb (hunk 3 above)
Comment (finalelement) [reply self_supervised_pretraining/ssl_train.ipynb:32]: I am assuming this is implying E --> e for the 'environment'.

File: self_supervised_pretraining/ssl_train.ipynb (hunk 4 above)
Comment (finalelement) [reply self_supervised_pretraining/ssl_train.ipynb:52]: Addressed

**Comment by finalelement** [comment]
Addressed all comments again, lets see if the tests work out this time.

**Comment by finalelement** [comment]
The 2 failing tests are from other prior notebooks. Let me know if more changes are needed to the SSL notebooks.

**Review by mingxin-zheng — APPROVED** [review]

**Comment by mingxin-zheng** [comment]
Thank you @finalelement . It looks good to me!

**Pull request merged**